Clarify which block's header hash the spec is referring to in the Tick Nonce Transition

Jimbo4350 opened the pull request.

Jimbo4350 committed:
Clarify which block's header hash the spec is referring to in the Tick
Nonce Transition

JaredCorduan reviewed (approved):
Thanks for cleaning this up @Jimbo4350 ! If you don't mind, I would like to add a commit to this PR that also adds a new item to the errata section about the fact that the consensus layer is actually using the penultimate block hash from the previous epoch. (I can make this second commit)

Commit:
errata item in shelley spec - penultimate hash

In the implemenation, the penultimate block hash of the previous epoch
is used to make the epoch nonce, not the last block hash as it
specified.

Jimbo4350 commented on eras/shelley/formal-spec/frontmatter.tex:
```diff
@@ -84,6 +85,7 @@ \section*{List of Contributors}
 Christian Lindgren,
 Yun Lu,
 Philipp Kant,
+Jordan Millar,
```
:partying_face:

Jimbo4350 reviewed (commented):
LGTM!
